### ISSUE TITLE
[FUND] Preconfigured, easy to install package for testing purposes

### ISSUE THREAD

**Issue by sebastian-meyer** [kitodo/kitodo-presentation#1176]
## Description

As part of software development and in preparation for new releases, tests have to be carried out time and again. Although there are now a large number of automated unit and integration tests, many functions can still only be tested manually.

For this purpose, a reproducible, preconfigured and easy-to-install appliance is to be created. This test environment should contain a yet-to-be-defined set of sample documents that are as varied as possible to enable comprehensive test scenarios. However, the appliance must not only be created statically once (e.g. as a Docker container), but must be able to be created dynamically using parameterizable software versions. For example, it must at least be possible to configure which PHP and TYPO3 versions and which Kitodo.presentation development status (branch or tag) is to be used when setting up the environment.

The set of sample documents could possibly be compiled by the Kitodo Community Board and made available by the respective owning institutions. Ideally, the test environment should also have a standard design to make it easier to carry out surface tests.

## Related Issues

#1177 is a prerequisite for this proposal.

## Expected Benefits of this Development

This will primarily benefit developers and release managers, but would also be an easy starting point for just getting familiar with Kitodo.Presentation or trying out the software.

## Estimated Costs and Complexity

I estimate the cost as **medium** to **high**.

**Comment by sebastian-meyer** [comment]
Votes: 8

**Comment by stweil** [comment]
Here are some additional requirements or questions:

1. Which hosts are supported (OS Linux / macOS, CPU Intel / ARM, OS releases)?
2. The package must allow tests of features which require Solr as well.
3. The package must fully support all implemented languages (currently German and English).

**Comment by stweil** [comment]
I added #1177 as a sub-issue because I also think that the issue here depends on that one. Hopefully everybody agrees – then there is no need to vote for #1177 separately.

And I suggest to add #923 and #1178 as sub-issues, too: the new preconfigured package should not show the deficits of the user interface which are discussed in #923, and of course it should also include a significantly improved documentation which covers the whole functionality of Kitodo.Presentation and which is up to date.

**Comment by sebastian-meyer** [comment]
Bundling #1177 is fine with me. I already mentioned this issue as a prerequisite in the initial issue description.

But I'd not recommend making this any bigger by including even more issues.

**Comment by sebastian-meyer** [comment]
An additional requirement for this is that it should be incorporated into the existing GitHub workflows, i. e. the package should be automatically deployable via GitHub Codespaces or similar services without needing a local environment.

**Comment by stweil** [comment]
Would it make sense to re-use https://github.com/kitodo/kitodo-demo/ or https://github.com/slub/dfg-viewer/ for the preconfigured package? This could reduce the required efforts and have a benefit for those repositories, too.

**Comment by sebastian-meyer** [comment]
Yes! In fact, @markusweigelt is already working on something like this for the DFG-Viewer and agreed to make his solution available to re-use for a more general approach for Kitodo.Presentation.
